inc/macro_utils/macro_utils.h

```diff
@@ -97,82 +97,136 @@ MU_IF(X, "true", "false") => "true"
 #define MU_DEFINE_ENUM_WITHOUT_INVALID(enumName, ...) typedef enum MU_C2(enumName, _TAG) { MU_FOR_EACH_1(MU_DEFINE_ENUMERATION_CONSTANT, __VA_ARGS__)} enumName; \
     typedef enum MU_C3(enum_value_metadata_, enumName, _TAG) { MU_FOR_EACH_1(MU_DEFINE_METADATA_ENUMERATION_VALUE, __VA_ARGS__) MU_C3(enum_value_metadata_, enumName, _VALUE_COUNT) = MU_COUNT_ARG(__VA_ARGS__)} MU_C2(enum_value_metadata_, enumName); \
-    extern const char* MU_C2(enumName,Strings)(enumName value); \
-    extern int MU_C2(enumName, _FromString)(const char* enumAsString, enumName* destination);
+    MU_DECLARE_ENUM_TO_STRING(enumName, __VA_ARGS__) \
 
 // this macro is a wrapper on top of MU_DEFINE_ENUM_WITHOUT_INVALID, adding an _INVALID value as the first enum value in the enum
 #define MU_DEFINE_ENUM(enumName, ...) \
     MU_DEFINE_ENUM_WITHOUT_INVALID(enumName, MU_C2(enumName, _INVALID), __VA_ARGS__)
 
+#define MU_INTERNAL_DEFINE_ENUM_VAR(enumName, enumValue) \
+    static enumName MU_C2(my_, enumValue);
+
+#define MU_INTERNAL_ASSIGN_ENUM_VALUE(argCount, enumValue) \
+    { \
+        if ((MU_C2(my_, enumValue)) == 0) \
+        { \
+            /* it is zero, either because not filled in or because it was explicit zero */ \
+            /* if it has an equal it cannot be implicit */ \
+            if (has_equal) \
+            { \
+                /* explicit zero */ \
+                current_enum_value = 0; \
+            } \
+            else \
+            { \
+                current_enum_value = previous_enum_value + 1; \
+            } \
+        } \
+        else \
+        { \
+            /* non zero value, thus it was filled in */ \
+            current_enum_value = MU_C2(my_, enumValue); \
+        } \
+        previous_enum_value = current_enum_value; \
+        if (current_enum_value == value) \
+        { \
+            return MU_TOSTRING(enumValue); \
+        } \
+    }
+
+#define MU_DECLARE_ENUM_TO_STRING(enumName, ...) \
+    const char* MU_C3(MU_, enumName, _ToString)(enumName value);
+
 #define MU_DEFINE_ENUMERATION_CONSTANT_AS_WIDESTRING(x) MU_C2(L, MU_TOSTRING(x)) , 
 #define MU_DEFINE_ENUMERATION_CONSTANT_AS_STRING(x) MU_TOSTRING(x) , 
+
+typedef enum MU_ENUM_VALUE_CONTAINS_EQUAL_TAG
+{
+    MU_ENUM_VALUE_CONTAINS_EQUAL_UNKNOWN,
+    MU_ENUM_VALUE_CONTAINS_EQUAL_YES,
+    MU_ENUM_VALUE_CONTAINS_EQUAL_NO
+} MU_ENUM_VALUE_CONTAINS_EQUAL;
+
+// variable definition indicating if an enum value label contains or not equal
+#define MU_INTERNAL_DEFINE_ENUM_HAS_EQ_VAR(argCount, enumValue) \
+    static MU_ENUM_VALUE_CONTAINS_EQUAL MU_C2(enum_value_has_equal_, argCount) = MU_ENUM_VALUE_CONTAINS_EQUAL_UNKNOWN; \
+
+// determine whether an enum value label has an equal or not
+#define MU_INTERNAL_COMPUTE_ENUM_HAS_EQ(argCount, enumValue) \
+    switch (MU_C2(enum_value_has_equal_, argCount)) \
+    { \
+        default: \
+        case MU_ENUM_VALUE_CONTAINS_EQUAL_UNKNOWN: \
+            if (strchr(MU_TOSTRING(MU_C2(my_, enumValue)), '=') != NULL) \
```

Comment:
>strchr [](http://example.com/codeflow?start=16&length=6)

why do we care to make this determination at runtime... every time. Can it be cached in any way?

Reply:
why I am at it - where is actually this macro used? Arrrgh, I need a fresh copy on my drive,.

Reply:
... errr... my visual studio doesn't compile this. or master! 16.9.5 it is :(. This is highly unusual. Becaue it compiles our EBS project... 

I will grab a VM in Azure. Hold on!

Reply:
... was missing git submodule update --init

Reply:
I came to the conclusion that this macro is not used, so it can be removed.

Reply:
I'm sooo sure that the tests should not pass! The macro should stay.
Highly suspect that there's some missing tests which should expose this utter horibleness.

---
In reply to: [632152688](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632152688) [](ancestors = 632152688)

Reply:
Suspicion has been confirmed.
A new iteration has been added that adds the cornercase which does need to make use of the has equal computation.

---
In reply to: [632161993](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632161993) [](ancestors = 632161993,632152688)

Reply:
Shall resolve this comment.

---
In reply to: [632143150](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632143150) [](ancestors = 632143150)

Reply:
Shall resolve this comment.

---
In reply to: [632151892](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632151892) [](ancestors = 632151892)

Reply:
It is cached by the virtue of 
    static MU_ENUM_VALUE_CONTAINS_EQUAL MU_C2(enum_value_has_equal_, argCount) = MU_ENUM_VALUE_CONTAINS_EQUAL_UNKNOWN; \


---
In reply to: [632137035](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632137035) [](ancestors = 632137035)

Reply:
Resolving this comment since probably a fresh copy was acquired. 

---
In reply to: [632138052](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632138052) [](ancestors = 632138052)